JavaScript/testing.md

```diff
@@ -0,0 +1,109 @@
+# Testing Philosophy
+
+>>> This was originally written for React Native, but the same principles can be applied to the web.
+
+Bugs come in many shapes and sizes. But in actuality, bugs can be broken down into a few categories.
+
+## Table of Contents
+
+- [Testing Philosophy](#Testing-Philosophy)
+  - [Table of Contents](#Table-of-Contents)
+  - [Bug Types](#Bug-Types)
+  - [Testing Methodologies](#Testing-Methodologies)
+    - [Testing for Style Bugs](#Testing-for-Style-Bugs)
+    - [Testing for Business Requirements](#Testing-for-Business-Requirements)
+    - [Testing for Data Flow](#Testing-for-Data-Flow)
+    - [Testing for Accidental Code Changes](#Testing-for-Accidental-Code-Changes)
+  - [Testing in Summary](#Testing-in-Summary)
+
+<a name="bug-types"></a>
+
+## Bug Types
+
+**1. Style bugs**
+
+When an implementation of a design does not match the design itself, it is a style bug. In mobile development, this usually happens when a certain screen size is not accounted for or when elements are not aligned properly. This also happens when developers need to interpret how designs should change when the screen size increases or decreases.
+
+**2. Business Requirements**
+
+When a flow is implemented, it usually determined by the requirements on the ticket itself. Sometimes these requirements are not explicitly written down or not followed to a T.
+
+**3. Data Flow**
+
+Coding is really just manipulating data. And on the front end, it's mostly just manipulating data that we either get from the backend or manipulating data structures that we create and store ourselves. Certain data problems that are fairly common are trying to manipulate data that is actually not the type we expect and trying to operate on data that is not defined.
```

Comment:
The examples mentioned at the end of this blip cover Type-related issues. Should mis-calculations or other unexpected behavior also be included here (or perhaps they'd be part of the Business Requirements bug)?

Reply:
I could see it going either way, but I think it's a great addition regardless of where we put it 👍